subsys/bluetooth/services/ots/ots.c

```diff
@@ -188,7 +188,7 @@ ssize_t ots_obj_name_write(struct bt_conn *conn,
 	}
 
 	/* Update real object name after no duplicate detected */
-	strcpy(ots->cur_obj->metadata.name, name);
+	utf8_lcpy(ots->cur_obj->metadata.name, name, CONFIG_BT_OTS_OBJ_MAX_NAME_LEN + 1);
```

Comment:
Hmm, outside the scope of this PR, but how does the stack know the size of the name buffer? 

The buffer is defined in the application, and the size is never communicated to the stack, nor does the stack require any explicit sizes.

Reply:
I believe it is an implied requirement that the name buffer provided by the application must be at least `CONFIG_BT_OTS_OBJ_MAX_NAME_LEN + 1` long. But I had the same thoughts as I was making this change.

This tangentially ties into our other conversations about who really owns the object metadata :)

Reply:
Indeed. At least from I can tell, that requirement is not written anywhere, and thus dangerous to assume. Even if it is required, it's a dangerous assumption, and the stack should not attempt to copy any data to a application buffer of unknown size.